app.js

```diff
@@ -103,7 +113,7 @@ io.on('connection', function(socket){
 
     var term;
     if (process.getuid() == 0) {
```

Comment:
I have a feeling you're talking about line 77:
```
if (opts.command) {
  command = opts.command;
}
```
 If we added the above line there it wouldn't work correctly, you're right. 

For line 115 though:
```
if (process.getuid() == 0) {
    term = pty.spawn(command, [], {
        name: 'xterm-256color',
        cols: 80,
        rows: 30
    });
} else {
    term = pty.spawn('ssh', [sshuser + sshhost, '-p', sshport, '-o', 'PreferredAuthentications=' + sshauth], {
        name: 'xterm-256color',
        cols: 80,
        rows: 30
    });
}
```
we're determining whether we have root access or not. Only the root user can run `/bin/login` so if we don't have it we fall back to `ssh`. The problem is that you only changed the section for `bin/login` and not `ssh`. So if Joe tries to run wetty and pass a command he will only get a password prompt from ssh.

If however we change line 115 to this:
```
if(process.getuid() == 0 || opts.command)
```
Joe's custom command would run.